diesel/src/mysql/query_builder/query_fragment_impls.rs

```diff
@@ -63,3 +69,71 @@ where
         Ok(())
     }
 }
+
+impl<T> QueryFragment<Mysql, crate::mysql::backend::MysqlOnConflictClause> for DoNothing<T>
+where
+    T: Table,
+    T::PrimaryKey: Column,
```

Comment:
Potentially unrelated question: the fact that we are restricting this implementation to just tables that have primary keys (at least that's what I think this is doing), does that mean that `diesel` doesn't require tables to have primary keys?

Reply:
Diesel requires tables to have primary keys. 
In fact this is just an "artificial" restriction because rustc cannot see that the `PrimaryKey` associated type of the `Table` trait implements `Column`